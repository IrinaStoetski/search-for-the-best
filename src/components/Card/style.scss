@import '../../styles/shared/style.scss';

.card {
  display: flex;
  flex-direction: column;
  padding: 8px;
  font-size: 18px;
  line-height: 25px;
  word-break: break-all;
  border: 1px solid $primary-color;
  border-radius: 6px;

  .card-title {
    font-weight: 700;
  }

  p {
    max-height: 500px;
    overflow: hidden; // for too big descriptions
  }

  &-image {
    align-self: center;
    width: 100px;
    height: 100px;
    margin-bottom: 8px;
    border-radius: 50%;

    img {
      @include rounded-image;
    }
  }

  a {
    color: $primary-color;
    font-size: 20px;
    text-decoration: none;
  }
}
