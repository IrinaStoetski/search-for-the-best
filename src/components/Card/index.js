import React from 'react';
import './style.scss';

const Card = ({
    avatar,
    name,
    description,
    link
}) => (

        <div className="card">
            <div className="card-image">
                <img src={avatar} />
            </div>
            <h3>
                <span className="card-title">
                   Name:&nbsp;</span>{name}
            </h3>
            <p>
              <span className="card-title">
                Description:&nbsp;</span>
                {description}
            </p>
            <a href={link} target="_blank">Visit owner's repository</a>
        </div>

    )

export default Card