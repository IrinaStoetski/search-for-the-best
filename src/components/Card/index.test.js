/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';

test('renders correctly card component', () => {
  const tree = renderer
    .create(<Card />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
