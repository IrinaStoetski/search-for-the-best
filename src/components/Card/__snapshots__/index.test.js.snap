// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly card component 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <img
      alt=""
    />
  </div>
  <h3>
    <span
      className="card-title"
    >
      Name: 
    </span>
  </h3>
  <p>
    <span
      className="card-title"
    >
      Description: 
    </span>
  </p>
  <a
    rel="noopener noreferrer"
    target="_blank"
  >
    Visit owner's repository
  </a>
</div>
`;
